### DISCUSSION TITLE
Euclidean subspace property

### DISCUSSION THREAD

**Discussion by StevenClontz** [pi-base/data#324] [Ideas]
Would it make sense to track which spaces are constructed (or are constructable) as subspaces of R^n? Such spaces are all separable metrizable of course. This occurred to me as potentially useful when discussing https://github.com/pi-base/data/issues/315

**Comment by prabau** [comment]
Could be handy.  "Embeddable in Euclidean space" or "Subspace of Euclidean space" or ... ?

**Comment by kulamateusz** [comment]
I wasn't sure were to post my suggestion, so it ends up here. I noticed that pi-base lacks the property 'nowhere compact' (each compact subset has empty interior) and I was wondering where to suggest creating it. I came up with some theorems about nowhere compact spaces:
weakly locally compact + non-empty -> ~nowhere compact
has an isolated point -> ~nowhere compact
~weakly locally compact + homogeneous -> nowhere compact
They are more or less trivial but thanks to them you can mark many spaces straight away.

More generally, how to become a contributor with mathematical content (theorems, properties, spaces) ?

**Reply by prabau** [reply]
Thanks for your interest.  We'll tell more about how to contribute.  But first one question.

One thing about pi-base is that it is not meant to be a place to discuss new topological notions.  Instead, it is meant to reflect notions that have already been published in the mathematical literature or in books, or at least have been discussed on sites like MathStackExchange of MathOverflow and are of interest to topologists.

Do you have any literature references for "nowhere compact"?

**Reply by prabau** [reply]
Regardless of whether we want to add 'nowhere compact' or not, we would be very happy to have you as a contributor.

You need to have a high level familiarity with git and the github workflow (commits, branches, etc).  See the help pages on github.  (But no need to get too deep in this.  Just high level is enough.)

See https://github.com/pi-base/data/wiki/Contributing.  I would recommend starting small to get used to the procedures.  There are many things that can currently be done to improve pi-base.  For example, various spaces have redundant asserted properties and some of that can be cleaned up.  Or adding new asserted properties (traits) to existing spaces.  So find something small that you can improve and create a PR (pull request) or first create an issue if you want to discuss the feasibility first.  The easiest is probably to use github.dev as explained in the web page above.  Just go for it.  If you get stuck, just ask and we'll guide you.
